test/rules/no-self-assign/default/test.ts.lint

```diff
@@ -0,0 +1,65 @@
+var a = a;
```

Comment:
Could you please add some cases with three variables in array and object spreads? Seems like that could be a cause of bugs.